addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4356,6 +4356,28 @@ export class OdooEditor extends EventTarget {
             if (selection.focusNode.nodeType === Node.ELEMENT_NODE) {
                 getDeepRange(this.editable, { selection, select: !isFocusContentEditable, correctTripleClick: !isFocusContentEditable });
             }
+            if (selection.focusNode.parentElement.classList.contains("o_text_highlight_item")) {
+                // We changed the `display` of `o_text_highlight_item` which is not a `block`
+                // so we should handle it separatly.
```

Comment:
What do you mean by "We changed the `display` of `o_text_highlight_item`" ? I see no change pertaining to that in this PR. Or you mean setting the class `o_text_highlight_item` on a P makes it behave like a non-block or something? I don't understand... ^^'

EDIT: I understood when I read the commit message, but the comment is unclear. It would be clearer in a passive tense, like:
```
// The `o_text_highlight_item` class applies `display: inline-block` which messes the selection.
```